packages/itwinui-react/src/core/Panels/Panels.tsx

```diff
@@ -0,0 +1,600 @@
+          currentPanelId,
+          changeActivePanel,
+          goBack,
+          triggers,
+          panelElements,
+          animations,
+        }}
+      >
+        <Box
+          ref={forwardedRef}
+          {...rest}
+          className={cx('iui-panel-wrapper', className)}
+        >
+          {children}
+        </Box>
+      </PanelsWrapperContext.Provider>
+    </PanelsInstanceProvider>
+  );
+}) as PolymorphicForwardRefComponent<'div', PanelsWrapperProps>;
+PanelsWrapper.displayName = 'Panels.Wrapper';
+
+const PanelsWrapperContext = React.createContext<
+  | {
+      activeId?: PanelsWrapperProps['activeId'];
+      currentPanelId: CurrentPanelId;
+      triggers: React.RefObject<
+        Map<string, { triggerId: string; panelId: string }>
+      >;
+      changeActivePanel: (props: {
+        newActiveId: CurrentPanelId;
+        ignoreControlledMode?: boolean;
+      }) => Promise<void>;
+      goBack: PanelsInstance['goBack'];
+      panelElements: React.RefObject<Record<string, HTMLElement | null>>;
+      animations: Record<string, React.CSSProperties[]> | undefined;
+    }
+  | undefined
+>(undefined);
+
+// ----------------------------------------------------------------------------
+
+type PanelProps = {
+  id: string;
+};
+
+/**
+ * Takes an `id` and the panel content. When the wrapper's `activeId` is set to this panel id, this panel is shown.
+ *
+ * @example
+ * <Panels.Panel id={panelIdRoot} as={List}>
+ *   <Surface.Header as={Panels.Header}>Root</Surface.Header>
+ *   <ListItem>
+ *     <Panels.Trigger for={panelIdMoreInfo}>
+ *       <ListItem.Action>More details</ListItem.Action>
+ *     </Panels.Trigger>
+ *   </ListItem>
+ * </Panels.Panel>
+ */
+const Panel = React.forwardRef((props, forwardedRef) => {
+  const { id: idProp, children, className, style, ...rest } = props;
+
+  const { currentPanelId, triggers, panelElements, animations } =
+    React.useContext(PanelsWrapperContext) || {};
+
+  const fallbackId = React.useId();
+  const id = idProp || fallbackId;
+
+  const [element, setElement] = React.useState<HTMLElement | null>(null);
+  if (panelElements?.current != null) {
+    panelElements.current[id] = element;
+  }
+
+  const isAnimating = animations != null;
+  const isHidden =
+    animations == null
+      ? id !== currentPanelId?.id
+      : !Object.keys(animations).includes(id);
+
+  const isInert = isHidden || isAnimating;
+
+  const associatedTrigger = !!id ? triggers?.current?.get(id) : undefined;
+
+  return (
+    <PanelContext.Provider
+      value={{
+        id,
+        associatedTrigger,
+      }}
+    >
+      <Box
+        ref={useMergedRefs(setElement, forwardedRef)}
+        id={id}
+        aria-labelledby={`${id}-header`}
+        tabIndex={-1}
+        {...rest}
+        className={cx('iui-panel', className, {
+          'iui-panel-visible': id === currentPanelId?.id,
+        })}
+        style={{
+          display: isHidden ? 'none' : undefined,
```

Comment:
Instead of setting `display` as an inline style, prefer setting the [`hidden`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) attribute and reinforcing it in CSS (see `Tabs` for example).